### ISSUE TITLE
https error

### ISSUE THREAD

**Issue by RunThem** [luarocks/luarocks#1360]
- **Platform:**
  - Linux

- **LuaRocks version:**
  - `luarocks --version`3.7

```
- sudo luarocks install  luaformatter

Error: LuaRocks 3.7.0 bug (please report at https://github.com/luarocks/luarocks/issues).
Arch.: linux-x86_64
/usr/share/lua/5.4/ssl/https.lua:66: bad argument #2 to 'method' (string expected, got light userdata)
stack traceback:
        [C]: in function 'socket.http.request'
        /usr/share/lua/5.4/ssl/https.lua:132: in function 'ssl.https.request'
        /usr/local/share/lua/5.4/luarocks/fs/lua.lua:729: in upvalue 'request'
        /usr/local/share/lua/5.4/luarocks/fs/lua.lua:837: in upvalue 'http_request'
        /usr/local/share/lua/5.4/luarocks/fs/lua.lua:904: in function 'luarocks.fs.lua.download'
        (...tail calls...)
        /usr/local/share/lua/5.4/luarocks/fetch.lua:154: in function 'luarocks.fetch.fetch_url'
        /usr/local/share/lua/5.4/luarocks/fetch.lua:50: in function 'luarocks.fetch.fetch_caching'
        /usr/local/share/lua/5.4/luarocks/manif.lua:108: in function 'luarocks.manif.load_manifest'
        /usr/local/share/lua/5.4/luarocks/search.lua:111: in function </usr/local/share/lua/5.4/luarocks/search.lua:101>
        (...tail calls...)
        /usr/local/share/lua/5.4/luarocks/search.lua:155: in function 'luarocks.search.search_repos'
        /usr/local/share/lua/5.4/luarocks/search.lua:241: in function 'luarocks.search.find_suitable_rock'
        /usr/local/share/lua/5.4/luarocks/search.lua:263: in function 'luarocks.search.find_rock_checking_lua_versions'
        /usr/local/share/lua/5.4/luarocks/cmd/install.lua:242: in function 'luarocks.cmd.install.command'
        (...tail calls...)
        [C]: in function 'xpcall'
        /usr/local/share/lua/5.4/luarocks/cmd.lua:663: in function 'luarocks.cmd.run_command'
        /usr/local/bin/luarocks:38: in main chunk
        [C]: in ?
```
Unable to download package

**Comment by hishamhm** [comment]
I believe this was [an issue with luasec and Lua 5.4](https://github.com/brunoos/luasec/issues/171). Upgrading LuaSec to 1.0.2 should work. Feel free to reopen if the problem persists with the latest LuaSec.

**Comment by filipmnowak** [comment]
@hishamhm, was it really fixed in `1.0.2`?

```sh
$ dpkg -l | grep lua-sec | awk '{print $2" | "$3}'
lua-sec:amd64 | 1.0.2-1
```

```sh
PRETTY_NAME="Ubuntu 22.04.1 LTS"
```
```sh
$ luarocks-5.4 --local install jsonschema

Error: LuaRocks 3.8.0 bug (please report at https://github.com/luarocks/luarocks/issues).
Arch.: linux-x86_64
/usr/share/lua/5.4/ssl/https.lua:66: bad argument #2 to 'method' (string expected, got light userdata)
stack traceback:
        [C]: in function 'socket.http.request'
        /usr/share/lua/5.4/ssl/https.lua:132: in function 'ssl.https.request'
        /usr/share/lua/5.4/luarocks/fs/lua.lua:729: in upvalue 'request'
        /usr/share/lua/5.4/luarocks/fs/lua.lua:837: in upvalue 'http_request'
        /usr/share/lua/5.4/luarocks/fs/lua.lua:904: in function 'luarocks.fs.lua.download'
        (...tail calls...)
        /usr/share/lua/5.4/luarocks/fetch.lua:154: in function 'luarocks.fetch.fetch_url'
        /usr/share/lua/5.4/luarocks/fetch.lua:50: in function 'luarocks.fetch.fetch_caching'
        /usr/share/lua/5.4/luarocks/manif.lua:108: in function 'luarocks.manif.load_manifest'
        /usr/share/lua/5.4/luarocks/search.lua:111: in function </usr/share/lua/5.4/luarocks/search.lua:101>
        (...tail calls...)
        /usr/share/lua/5.4/luarocks/search.lua:155: in function 'luarocks.search.search_repos'
        /usr/share/lua/5.4/luarocks/search.lua:241: in function 'luarocks.search.find_suitable_rock'
        /usr/share/lua/5.4/luarocks/search.lua:263: in function 'luarocks.search.find_rock_checking_lua_versions'
        /usr/share/lua/5.4/luarocks/cmd/install.lua:242: in function 'luarocks.cmd.install.command'
        (...tail calls...)
        [C]: in function 'xpcall'
        /usr/share/lua/5.4/luarocks/cmd.lua:666: in function 'luarocks.cmd.run_command'
        /usr/bin/luarocks:35: in main chunk
        [C]: in ?
```

**Comment by hishamhm** [comment]
@filipmnowak I don't know for sure, but luasec is currently at 1.2.0. I just enabled the latest luasec in our test suite, and it looks like things are working: https://github.com/luarocks/luarocks/pull/1434
